compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/ConvertConv2DToIm2ColOp.cpp

```diff
@@ -36,12 +35,40 @@ static Value createMul(Location loc, Value x, Value y, OpBuilder &builder) {
   return builder.create<arith::MulFOp>(loc, x, y);
 }
 
+// TODO : Upstream utility that does this pruning is broken for LinalgOp. Drop
+// this if that gets fixed.
+static SmallVector<NamedAttribute> getPrunedAttributeList(linalg::LinalgOp op) {
+  const StringLiteral memoAttr =
+      linalg::LinalgDialect::kMemoizedIndexingMapsAttrName;
+  SmallVector<NamedAttribute> prunedAttributeList;
+  for (auto attr : op->getDiscardableAttrs()) {
+    if (attr.getName() != memoAttr) {
+      prunedAttributeList.push_back(attr);
+    }
+  }
+  return prunedAttributeList;
+}
+
+// Helper to convert a shape into basis for im2col op.
+static SmallVector<int64_t> getBasisFromShape(ArrayRef<int64_t> shape) {
+  SmallVector<int64_t> basis(shape.size());
+  int64_t cummulativeProduct = 1;
+  for (int i = shape.size() - 1; i >= 0; --i) {
+    cummulativeProduct *= shape[i];
+    basis[i] = cummulativeProduct;
+  }
+  // Shift left with innermost basis one.
+  basis.push_back(1);
+  // Drop the outermost basis.
+  return llvm::to_vector(llvm::ArrayRef(basis).drop_front());
```

Comment:
nit: I think if you swap the order of L57-58, then you don't need to shift the basis/drop the outermost.
```suggestion
  for (int i = shape.size() - 1; i >= 0; --i) {
    basis[i] = cummulativeProduct;
    cummulativeProduct *= shape[i];
  }
  return basis;
```